sql/src/main/java/io/crate/analyze/validator/GroupBySymbolValidator.java

```diff
@@ -40,7 +41,8 @@ public static void validate(Symbol symbol) throws IllegalArgumentException, Unsu
     }
 
     private static void validateDataType(Symbol symbol) {
-        if (!DataTypes.PRIMITIVE_TYPES.contains(symbol.valueType())) {
+        if (!(symbol.symbolType() == SymbolType.LITERAL && symbol.valueType() == DataTypes.UNDEFINED) &&
+            !DataTypes.PRIMITIVE_TYPES.contains(symbol.valueType())) {
```

Comment:
ah, i think i missed it you fixed the null case (pls add a test), however i think this is not enough.
i think it is wrong to check the data type here, since the constraint we have here is the type of index not the datatype. since literals have no index and are not affected in the collect phase we can support them all.

so imho any literal should work, right? e.g:

```
cr> select array[1,2], count(*) from sys.cluster group by 1;
SQLActionException[SQLParseException: Cannot GROUP BY '_array(1, 2)': invalid data type 'long_array']
cr> select distinct array[1,2];
SQLActionException[SQLParseException: Cannot GROUP BY '_array(1, 2)': invalid data type 'long_array']
```